wordpress-mobile/WordPress-Android
# Show self-hosted sign-in screen for reserved WP.com usernames

Fixes #4684. The sign-in screen will now recognize reserved WordPress.com usernames and send the user to the self-hosted screen immediately if the username can't be for WordPress.com.

iOS counterpart PR (adds more names and `*wordpress*` to pre-existing check): https://github.com/wordpress-mobile/WordPress-iOS/pull/6128

To test:
1. Enter any of the reserved usernames in the user field
   1. Any of `admin`, `administrator`, `invite`, `main`, `root`, `web`, `www` should trigger the self-hosted screen
   2. Any username containing `wordpress` in any configuration should trigger
   3. Any _email_ containing `wordpress` should _not_ trigger the self-hosted screen<sup><a name="footnotelink1" href="#footnote1">1</sup></a>
2. Switch to the password field -> the URL field should appear
3. Confirm that you can complete self-hosted sign on from here

Needs review: @kwonye for consistency with iOS 😀

<a name="footnote1" href="#footnotelink1"><sup>1</sup></a> This will be redundant if we implement magic links permanently, since the email is checked in advance, and the self-hosted screen should be displayed no matter what if the email isn't registered on WordPress.com
